### PR TITLE
Test unbundled ESM versions of driver adapters

### PR DESCRIPTION
**Pull request by SevInf**
Planetscale ESM test will fail until prisma/prisma#21962 is merged.

### CODE REVIEW

File: driver-adapters/planetscale-node-basic-esm/package.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "planetscale-node-basic",
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "test": "NODE_OPTIONS=--experimental-vm-modules jest index.test.mjs"
```
Comment (Jolg42) [inline driver-adapters/planetscale-node-basic-esm/package.json:6]: Note: that's the important part here.

**Comment by Jolg42** [comment]
@SevInf nice one!

We can't merge as is though, because this breaks the isolation of these tests.
Before each had its own database, now, the additional tests are sharing the database with previous ones leading to some issues like:
https://github.com/prisma/ecosystem-tests/actions/runs/6876937563/job/18703691194?pr=4247#step:8:236
```
    Unique constraint failed on the fields: (`email`)
```

2 solutions for me
- we make it so that it runs on a separate db
	- it's more work to do and maintain, more env vars ...
	- we only need to test that ESM works, I don't think duplicating the test brings more value than next option below
- we could change the tests
	- we could maybe only test that a query can execute successfully (not checking the return value, and remove all other queries)

**Comment by SevInf** [comment]
@Jolg42 should be good now, waiting for `dev` branch update to see planetscale fixed.

File: driver-adapters/neon-node-basic-esm/index.test.mjs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// @ts-check
+import { test, expect, jest } from '@jest/globals'
+import { handler } from './index.mjs'
+
+jest.setTimeout(15_000)
+
```
Comment (Jolg42) [inline driver-adapters/neon-node-basic-esm/index.test.mjs:6]:
```suggestion
```

File: driver-adapters/planetscale-node-basic-esm/index.test.mjs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// @ts-check
+import { test, expect, jest } from '@jest/globals'
+import { handler } from './index.mjs'
+
+jest.setTimeout(15_000)
+
```
Comment (Jolg42) [inline driver-adapters/planetscale-node-basic-esm/index.test.mjs:6]:
```suggestion
```

File: driver-adapters/turso-node-basic-esm/package.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "turso-node-basic",
```
Comment (Jolg42) [inline driver-adapters/turso-node-basic-esm/package.json]:
```suggestion
  "name": "turso-node-basic-esm",
```

File: driver-adapters/neon-node-basic-esm/package.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "neon-node-basic",
```
Comment (Jolg42) [inline driver-adapters/neon-node-basic-esm/package.json]:
```suggestion
  "name": "neon-node-basic-esm",
```

File: driver-adapters/planetscale-node-basic-esm/package.json [hunk 6]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "planetscale-node-basic",
```
Comment (Jolg42) [inline driver-adapters/planetscale-node-basic-esm/package.json]:
```suggestion
  "name": "planetscale-node-basic-esm",
```

File: driver-adapters/planetscale-node-basic-esm/prepare.sh [hunk 7]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+set -eu
+
+export PRISMA_TELEMETRY_INFORMATION='ecosystem-tests driver-adapters planetscale-node-basic'
```
Comment (Jolg42) [inline driver-adapters/planetscale-node-basic-esm/prepare.sh]:
```suggestion
export PRISMA_TELEMETRY_INFORMATION='ecosystem-tests driver-adapters planetscale-node-basic-esm'
```

File: driver-adapters/turso-node-basic-esm/prepare.sh [hunk 8]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+set -eu
+
+export PRISMA_TELEMETRY_INFORMATION='ecosystem-tests driver-adapters turso-node-basic'
```
Comment (Jolg42) [inline driver-adapters/turso-node-basic-esm/prepare.sh]:
```suggestion
export PRISMA_TELEMETRY_INFORMATION='ecosystem-tests driver-adapters turso-node-basic-esm'
```

File: driver-adapters/neon-node-basic-esm/prepare.sh [hunk 9]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+set -eu
+
+export PRISMA_TELEMETRY_INFORMATION='ecosystem-tests driver-adapters neon-node-basic'
```
Comment (Jolg42) [inline driver-adapters/neon-node-basic-esm/prepare.sh]:
```suggestion
export PRISMA_TELEMETRY_INFORMATION='ecosystem-tests driver-adapters neon-node-basic-esm'
```

**Review by Jolg42 — APPROVED** [review]
See small suggestions

**Comment by SevInf** [comment]
Tests passed after `5.7.0-dev.6` update, merging now.

**Pull request merged**